src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -194,22 +244,17 @@ Object* FrozenObjectSegment::TryAllocateObject(PTR_MethodTable type, size_t obje
         m_SizeCommitted += FOH_COMMIT_SIZE;
     }
 
-    INDEBUG(m_ObjectsCount++);
-
     Object* object = reinterpret_cast<Object*>(m_pCurrent);
     object->SetMethodTable(type);
 
     m_pCurrent += objectSize;
 
-    // Notify GC that we bumped the pointer and, probably, committed more memory in the reserved part
-    GCHeapUtilities::GetGCHeap()->UpdateFrozenSegment(m_SegmentHandle, m_pCurrent, m_pStart + m_SizeCommitted);
-
     return object;
 }
 
 Object* FrozenObjectSegment::GetFirstObject() const
 {
-    if (m_pStart + sizeof(ObjHeader) == m_pCurrent)
+    if (m_pStart + sizeof(ObjHeader) == m_pCurrentRegistered)
```

Comment:
Why do we need to switch to `m_pCurrentRegistered` here? 

We are under `m_Crst` lock when enumerating the objects.